File: Minecraft2D/Rendering/OpaqueRichTextBox.cs
using System.Drawing;
using System.Windows.Forms;

namespace Minecraft2D
{
    public class OpaqueRichTextBox : RichTextBox
    {
        private static readonly Color DefaultBackground = Color.Transparent;
        private Image TransparentImage;

        public OpaqueRichTextBox()
        {
            SetStyle(ControlStyles.AllPaintingInWmPaint, true);
            SetStyle(ControlStyles.Opaque, true);
            SetStyle(ControlStyles.SupportsTransparentBackColor, true);
            BackColor = DefaultBackground;
        }

        public override Color BackColor
        {
            get
            {
                return base.BackColor;
            }

            set
            {
                base.BackColor = value;
            }
        }

        public Image Image
        {
            get
            {
                return TransparentImage;
            }

            set
            {
                TransparentImage = value;
                Invalidate();
            }
        }

        protected override CreateParams CreateParams
        {
            get
            {
                var cp = base.CreateParams;
                cp.ExStyle = cp.ExStyle | 0x20;
                return cp;
            }
        }

        // Infrastructure to cause the default background to be transparent
        public bool ShouldSerializeBackColor()
        {
            return BackColor == DefaultBackground;
        }

        // Infrastructure to cause the default background to be transparent
        public void ResetBackground()
        {
            BackColor = DefaultBackground;
        }
    }
}

File: Minecraft2D/Rendering/OpaqueRichTextBox.vb
Public Class OpaqueRichTextBox
    Inherits RichTextBox

    Private Shared ReadOnly DefaultBackground As Color = Color.Transparent
    Private TransparentImage As Image

    Public Sub New()
        Me.SetStyle(ControlStyles.AllPaintingInWmPaint, True)
        Me.SetStyle(ControlStyles.Opaque, True)
        Me.SetStyle(ControlStyles.SupportsTransparentBackColor, True)
        BackColor = DefaultBackground
    End Sub

    Public Overrides Property BackColor() As System.Drawing.Color
        Get
            Return MyBase.BackColor
        End Get
        Set(ByVal value As System.Drawing.Color)
            MyBase.BackColor = value
        End Set
    End Property

    Public Property Image() As Image
        Get
            Return TransparentImage
        End Get
        Set(ByVal value As Image)
            TransparentImage = value
            Invalidate()
        End Set
    End Property

    Protected Overrides ReadOnly Property CreateParams() As System.Windows.Forms.CreateParams
        Get
            Dim cp As CreateParams = MyBase.CreateParams
            cp.ExStyle = cp.ExStyle Or &H20
            Return cp
        End Get
    End Property

    ' Infrastructure to cause the default background to be transparent
    Public Function ShouldSerializeBackColor() As Boolean
        Return BackColor = DefaultBackground
    End Function

    ' Infrastructure to cause the default background to be transparent
    Public Sub ResetBackground()
        BackColor = DefaultBackground
    End Sub

End Class